### DISCUSSION TITLE
Does Thrust support cuda-aware mpi for multi gpu executions?

### DISCUSSION THREAD

**Discussion by manasi-t24** [NVIDIA/cccl#1039] [Thrust]
Hello,
I am going through Nvidia blogs regarding optimization of GPU code.
I came across this post https://devblogs.nvidia.com/introduction-cuda-aware-mpi/ about using cuda-aware mpi to speed up the data transfer between two GPU's (either on the same node or on different nodes). 
I have an application RapidCFD which uses thrust libraries for multi GPU solving of a system of linear equations. When I profiled the application, I saw that a lot of time is being consumed in data transfers. 
Since, the data transfer details are hidden to the user because thrust library is used, I want to know if thrust employs all the data transfer optimizations mentioned by Nvidia in their blog posts, like the above one about direct data transfer between 2 GPU's using cuda aware mpi and this one [(https://devblogs.nvidia.com/how-optimize-data-transfers-cuda-cc/)](url) about allocating storage in the pinned memory.

Any help would be appreciated.
Regards,
Manasi

**Comment by jrhemstad** [comment]
Thrust algorithms do not yet support using multi-GPUs, and likewise they do not use MPI or any communication library underneath. The only data transfers in Thrust would be host <-> device transfers for a single GPU.
